packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py

```diff
@@ -66,36 +66,51 @@ async def push(
         return await _push_file(user_id, project_id, node_uuid, archive_file_path, None)
 
 
-async def _pull_file(user_id: int, project_id: str, node_uuid: str, file_path: Path):
+async def _pull_file(
+    user_id: int,
+    project_id: str,
+    node_uuid: str,
+    file_path: Path,
+    save_to: Optional[Path] = None,
+):
+    destination_path = file_path if save_to is None else save_to
     s3_object = _create_s3_object(project_id, node_uuid, file_path)
     log.info("pulling data from %s to %s...", s3_object, file_path)
     downloaded_file = await filemanager.download_file_from_s3(
         user_id=user_id,
         store_id="0",
         s3_object=s3_object,
-        local_folder=file_path.parent,
+        local_folder=destination_path.parent,
     )
-    if downloaded_file != file_path:
-        if file_path.exists():
-            file_path.unlink()
-        move(f"{downloaded_file}", file_path)
-    log.info("%s successfuly pulled", file_path)
+    if downloaded_file != destination_path:
+        if destination_path.exists():
+            destination_path.unlink()

```

Comment:
FYI: Since 3.8, now we can write this in a single line SEE https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink
```python
destination_path.unlink(missing_ok=True)
```

Reply:
uhhh, nice one!